packages/core/src/sync-realtime/index.ts

```diff
@@ -516,7 +548,9 @@ export const createRealtimeSync = (
               );
               const pendingBlocks = await Promise.all(
                 missingBlockRange.map((blockNumber) =>
-                  _eth_getBlockByNumber(args.requestQueue, { blockNumber }),
+                  _eth_getBlockByNumber(args.requestQueue, {
+                    blockNumber,
+                  }).then((block) => extract(block)),
```

Comment:
We could rename `extract` to something like `fetchRequiredBlockData` or `fetchFullBlockData` if we like that name